### PR TITLE
fix: re-enable services after snap disable/enable cycle

### PR DESCRIPTION
**Pull request by johnramsden**
# Description

When `snap disable microceph` is run followed by `snap enable microceph`, only the main daemon service gets re-enabled. All other services (mon, mgr, mds, osd, rgw, etc.) remain disabled because they use `install-mode: disable` in snapcraft.yaml.

On daemon startup, query the database for services registered on this host and re-enable any that are inactive. OSD is always re-enabled since it is started unconditionally at bootstrap.

A serviceStartMu lock is used to prevent the re-enablement goroutine from racing with bootstrap/join, which also start services via snapctl. Without it, concurrent `snapctl start --enable` calls cause "service-control change in progress" errors.

Note: For consistency I also changed a few close by context deadlines to the slightly more correct pattern: 

```
- time.Sleep(time.Minute)                                              
+ select {                                                             
+     case <-ctx.Done():                                                   
+         return                                                             
+     case <-time.After(time.Minute):                                      
+ }
```

Also fixed bug where `getServicesForHost` overrode a param.

Fixes: #525

## Type of change

Delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Clean code (code refactor, test updates; does not introduce functional changes)

## How has this been tested?

Added integration test to confirm services are the same before/after enable/disable.

I decided unit tests did not make sense here since this feature is tightly coupled with the database and other areas. I did not determine it to be useful giving the amount of mocking that would be required.

## Contributor checklist

Please check that you have:

- [x] self-reviewed the code in this PR
- [x] added code comments, particularly in less straightforward areas
- [x] checked and added or updated relevant documentation
- [x] checked and added or updated relevant release notes
- [x] added tests to verify effectiveness of this change

### CODE REVIEW

File: tests/scripts/actionutils.sh [hunk 1]
```diff
@@ -1630,6 +1630,67 @@ function verify_mount_check() {
...
+    echo "--- snap services before disable ---"
+    snap services microceph
+
+    # Record which services are both enabled and active before disable.
+    # Only these should come back — transiently started services (active
+    # but Startup=disabled) are not DB-registered and won't be restored.
+    local services_before
+    services_before=$(snap services microceph | awk '$2 == "enabled" && $3 == "active" {print $1}')
+    local count_before
+    count_before=$(echo "$services_before" | wc -l)
```
Comment (sabaini) [inline tests/scripts/actionutils.sh]: Nit: tiny edge case here if there's no services before
```
echo "" | wc -l
1
```
Maybe sth like `count_before=$(printf '%s' "$services_before" | sed '/^$/d' | wc -l)`

Comment (johnramsden) [reply tests/scripts/actionutils.sh]: added https://github.com/canonical/microceph/pull/691/commits/dfc179f851eb88ac4bb9a6997aba07fdb9807d06
